test/mitmproxy/tservers.py

```diff
@@ -107,11 +103,23 @@ def port(self):
     def tlog(self):
         return self.tmaster.logs
 
+    def shutdown(self):
+        self.tmaster.shutdown()
+
     def run(self):
+        self.event_loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(self.event_loop)
+        self.tmaster = self.masterclass(self.options)
+        self.tmaster.addons.add(core.Core())
+        self.name = "ProxyThread (%s:%s)" % (
+            self.tmaster.server.address[0],
+            self.tmaster.server.address[1],
+        )
```

Comment:
I think we have a `human_address` function for this somewhere...?

Reply:
Done.